lib/usdUfe/ufe/UsdClipboard.cpp

```diff
@@ -0,0 +1,129 @@
+//
+// Copyright 2024 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "UsdClipboard.h"
+
+#include <pxr/usd/sdf/layer.h>
+#include <pxr/usd/usd/usdFileFormat.h>
+#include <pxr/usd/usd/usdcFileFormat.h>
+#include <pxr/usd/usdShade/connectableAPI.h>
+
+#if (__cplusplus < 201703L)
+#error "Compiling ClipboardHandler requires C++17"
+#endif
+#include <filesystem> // Requires C++17
+
+namespace USDUFE_NS_DEF {
+
+// Ensure that UsdClipboard is properly setup.
+static_assert(!std::is_copy_constructible<UsdClipboard>::value);
+static_assert(!std::is_copy_assignable<UsdClipboard>::value);
+static_assert(!std::is_move_constructible<UsdClipboard>::value);
+static_assert(!std::is_move_assignable<UsdClipboard>::value);
+
+UsdClipboard::UsdClipboard()
+{
+    setClipboardPath(std::filesystem::temp_directory_path().string());
+    setClipboardFileFormat(PXR_NS::UsdUsdcFileFormatTokens->Id.GetText()); // Default = binary
+}
+
+UsdClipboard::~UsdClipboard() { cleanClipboard(); }
+
+void UsdClipboard::setClipboardData(const PXR_NS::UsdStageWeakPtr& clipboardData)
+{
+    // Note: if a clipboard file already exists, it automatically gets overridden, so there is no
+    // need to clear it.
+    // Note: export the root layer directly as the stage export will flatten which removes
+    //       variant sets, payloads, etc.
+    PXR_NS::SdfFileFormat::FileFormatArguments args;
+    args[PXR_NS::UsdUsdFileFormatTokens->FormatArg] = _clipboardFileFormat;
+    if (!clipboardData->GetRootLayer()->Export(_clipboardFilePath, "UsdUfe clipboard", args)) {
```

Comment:
In LDX the export was done using the method on the UsdStage which flattens, which removes variant sets, etc. The export method on the layer does not flatten.

Reply:
I actually already have a code comment just above that explains this.